osbenchmark/workload/params.py

```diff
@@ -1186,45 +1179,68 @@ def params(self):
         self.percent_completed = self.current / self.total
         return self.query_params
 
-    def _build_vector_search_query_body(self, vector, efficient_filter=None) -> dict:
-        """Builds a vector search request that can be used to execute an approximate nearest
+    def _build_vector_search_query_body(self, vector, efficient_filter=None, filter_type=None, filter_body=None) -> dict:
+        """Builds a k-NN request that can be used to execute an approximate nearest
         neighbor search against a k-NN plugin index
         Args:
             vector: vector used for query
-            efficient_filter: efficient filter used for query
         Returns:
             A dictionary containing the body used for search query
         """
-        query = {}
-        if self.query_type == self.KNN_QUERY_TYPE:
-            query.update({
-                "k": self.k,
-            })
-        elif self.query_type == self.MIN_SCORE_QUERY_TYPE:
-            query.update({
-                "min_score": self.score,
-            })
-        elif self.query_type == self.MAX_DISTANCE_QUERY_TYPE:
-            query.update({
-                "max_distance": self.distance,
-            })
-        else:
-            raise Exception("Unknown query type [%s]" % self.query_type)
-
-        query.update({
+        query = {
             "vector": vector,
-        })
-
+            "k": self.k,
+        }
         if efficient_filter:
             query.update({
                 "filter": efficient_filter,
             })
-        return {
+
+        knn_search_query = {
             "knn": {
                 self.field_name: query,
             },
         }
 
+        if self.is_nested:
+            outer_field_name, _inner_field_name = self.get_split_fields()
+            return {
+                "nested": {
+                    "path": outer_field_name,
+                    "query": knn_search_query
+                }
+            }
+
+        if filter_type and not efficient_filter and not filter_type == "post_filter":
+            return self._knn_query_with_filter(vector, knn_search_query, filter_type, filter_body)
+
+        return knn_search_query
+
+    def _knn_query_with_filter(self, vector, knn_query, filter_type, filter_body) -> dict:
+        if filter_type == "script":
+            return {
+                "script_score": {
+                    "query": {"bool": {"filter": filter_body}},
+                    "script": {
+                        "source": "knn_score",
```

Comment:
Why use knn script score here?

Reply:
I added it in OSB since it's mentioned in the [filters documentation](https://opensearch.org/docs/latest/search-plugins/knn/filter-search-knn/#scoring-script-filter) but I haven't written a workload yet. Is Knn script score something that we would like to benchmark?

Reply:
I spoke with @VijayanB and it seems like @navneet1v is adding a feature in 2.17 where exact search can be performed via a knn query (as opposed to the script score query). This change should be much easier to benchmark than exact search via a `script_score` query since it would just require a new workload without any OSB runner changes. I'll remove this script score code in the next revision.

Reply:
@finnroblin but we will still need script score based benchmarks.

Reply:
@jmazanec15, @finnroblin do we not want to add script score based benchmarks?

Reply:
We probably shoud have this as well. But I was thinking that exact search should use new method for implementing it